test/extensions/transport_sockets/proxy_protocol/proxy_protocol_integration_test.cc

```diff
@@ -0,0 +1,138 @@
+#include "envoy/config/core/v3/base.pb.h"
+#include "envoy/config/core/v3/proxy_protocol.pb.h"
+#include "envoy/extensions/transport_sockets/proxy_protocol/v3/upstream_proxy_protocol.pb.h"
+
+#include "test/integration/integration.h"
+
+namespace Envoy {
+namespace {
+
+class ProxyProtocolIntegrationTest : public testing::TestWithParam<Network::Address::IpVersion>,
+                                     public BaseIntegrationTest {
+public:
+  ProxyProtocolIntegrationTest()
+      : BaseIntegrationTest(GetParam(), ConfigHelper::tcpProxyConfig()) {}
+
+  void TearDown() override {
+    test_server_.reset();
+    fake_upstreams_.clear();
+  }
+
+  void setVersion(envoy::config::core::v3::ProxyProtocolConfig_Version version) {
+    version_ = version;
+  }
+
+  void initialize() override {
+    config_helper_.addConfigModifier([this](envoy::config::bootstrap::v3::Bootstrap& bootstrap) {
+      auto* transport_socket =
+          bootstrap.mutable_static_resources()->mutable_clusters(0)->mutable_transport_socket();
+      transport_socket->set_name("envoy.transport_sockets.upstream_proxy_protocol");
+      envoy::config::core::v3::TransportSocket raw_transport_socket;
+      raw_transport_socket.set_name("envoy.transport_sockets.raw_buffer");
+      envoy::config::core::v3::ProxyProtocolConfig proxy_proto_config;
+      proxy_proto_config.set_version(version_);
+      envoy::extensions::transport_sockets::proxy_protocol::v3::ProxyProtocolUpstreamTransport
+          proxy_proto_transport;
+      proxy_proto_transport.mutable_transport_socket()->MergeFrom(raw_transport_socket);
+      proxy_proto_transport.mutable_config()->MergeFrom(proxy_proto_config);
+      transport_socket->mutable_typed_config()->PackFrom(proxy_proto_transport);
+    });
+    BaseIntegrationTest::initialize();
+  }
+
+private:
+  envoy::config::core::v3::ProxyProtocolConfig_Version version_;
+};
+
+INSTANTIATE_TEST_SUITE_P(IpVersions, ProxyProtocolIntegrationTest,
+                         testing::ValuesIn(TestEnvironment::getIpVersionsForTest()),
+                         TestUtility::ipTestParamsToString);
+
+// Test sending proxy protocol v1
+TEST_P(ProxyProtocolIntegrationTest, TestV1ProxyProtocol) {
+  setVersion(envoy::config::core::v3::ProxyProtocolConfig::V1);
+  initialize();
+
+  auto listener_port = lookupPort("listener_0");
```

Comment:
Given you already have a custom initialize, how about you make tcp_client and fake_upstream_connection class variables (with trailing underscores) and move this block to initialize()?

Reply:
I didn't quite do this one. The health check test doesn't create a client connection so I left that in individual tests. I moved the `FakeRawConnectionPtr` to a class variable though.